File: C++/SelectionSort/SelectionSort/Student.cpp
//Student Class implementation file
//Damian Zylski
//09/13/2020

#include <iostream>
#include <string>
#include "Person.h"
#include "Student.h"
using namespace std;

//constructors
Student::Student() : Person()
{
	studentID = 0;
	major = "Blank";
	level = "Blank";
}

Student::Student(int id, string m, string l, string n) : Person(n)
{
	studentID = id;
	major = m;
	level = l;
}

Student::~Student()
{

}

//Accessors
int Student::getID()
{
	return studentID;
}

string Student::getMajor()
{
	return major;
}

string Student::getLevel()
{
	return level;
}

//Mutators
void Student::setID(int id)
{
	studentID = id;
}

void Student::setMajor(string m)
{
	major = m;
}

void Student::setLevel(string l)
{
	level = l;
}

//write Output
void Student::writeOutput()
{
	Person::writeOutput();
	cout << "Student ID: " << studentID << endl;
	cout << "Major: " << major << endl;
	cout << "Level: " << studentID << endl << endl;
}

//Equal
bool Student::sameStudent(Student s)
{
	if (this->getID() == s.getID())
	{
		return true;
	}
	else
	{
		return false;
	}
}

//compareTo
int Student::compareTo(Student s)
{
	if (this->getID() > s.getID())
	{
		return 1;
	}
	else if (this->getID() < s.getID())
	{
		return -1;
	}
	else if (this->getID() == s.getID())
	{
		return 0;
	}
	else
	{
		return 0;
	}
}

string Student::toString()
{
	return "" + studentID;
}

File: Java/PeopleWork/src/peoplework/Student.java
package peoplework;

/**
 * Programmer: Damian Zylski
 * Project:    PeopleWork
 * Class:      Student
 * Date:       08/10/2020
 * 
 */

public class Student extends Person implements Comparable <Student>
{
    //Attributes
    private int studentID;
    private String level;
    private String major;
    
    //Constructors
    public Student()
    {
        super();
        studentID = 000000;
        level = "blank";
        major = "none";
    }
    
    public Student(int id, String m, String l, String n)
    {
        super(n);
        studentID = id;
        level = l;
        major = m;
    }
    
    //Setters
    public void setID(int id)
    {
        studentID = id;
    }
    public void setLevel(String l)
    {
        level = l;
    }
    public void setMajor(String m)
    {
        major = m;
    }
    //Getters
    public int getID()
    {
        return studentID;
    }
    public String getLevel()
    {
        return level;
    }
    public String getMajor()
    {
        return major;
    }
    //Output
    @Override
    public void writeOutput()
    {
        super.writeOutput();
        System.out.println("Student ID: " + studentID);
        System.out.println("Level: " + level);
        System.out.println("Major: " + major);
        System.out.println("");
    }
    //check if Equal
    public boolean isSame(Student s)
    {
        //Student ID numbers are unique
        if(super.hasSameName(s) && this.studentID == s.studentID)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    
    //compareTo
    public int compareTo(Student s)
    {
        if(this.studentID > s.studentID)
        {
            return 1; //greater than
        }
        else if(this.studentID < s.studentID)
        {
            return -1; //less than
        }
        else if(this.studentID == s.studentID)
        {
            return 0; //equal
        }
        else
        {
            return 0; //default
        }
    }
    //toString
    public String toString()
    {
        return "" + studentID;
    }
    
}
